pkg/validate/openshiftcluster_validatedynamic.go

```diff
@@ -175,7 +178,24 @@ func (dv *openShiftClusterDynamicValidator) Dynamic(ctx context.Context) error {
 			return err
 		}
 	} else {
-		// PlatformWorkloadIdentity and ClusterMSIIdentity Validation
+		//ClusterMSI Validation
+		cmsiDynamic := dynamic.NewValidator(
+			dv.log,
+			dv.env,
+			dv.env.Environment(),
+			dv.subscriptionDoc.ID,
+			dv.fpAuthorizer,
+			nil,
+			dynamic.AuthorizerClusterUserAssignedIdentity,
+			dv.clusterMSICredential,
+			pdpClient,
+		)
+		err = cmsiDynamic.ValidateClusterUserAssignedIdentity(ctx, dv.oc.Properties.PlatformWorkloadIdentityProfile.PlatformWorkloadIdentities, dv.roleDefinitions)
+		if err != nil {
+			return err
+		}
```

Comment:
It seems to be only used to run `ValidateClusterUserAssignedIdentity`.
If `spDynamic` has `dv.clusterMSICredential`, can `spDynamic` run `ValidateClusterUserAssignedIdentity`, and can we remove `cmsiDynamic`?

Reply:
We are performing dv.clusterMSICredential as part of `cmsiDynamic` because we need  dv.clusterMSICredential to get the EntraID access token and oid for the jwttoken present in dv.clusterMSICredential.
But all the validations done by `spDynamic` are without the jwttoken, it has been broken up into two different dynamic objects to separate their way of creating the pdpClient's authorization request.

I would keep them separated as they are validations for 2 different type of identities, one is Cluster MSI(work with access token) and other being all the Platform Workload Identities(oid need to be passed, no jwttoken exists).